stl/inc/ranges

```diff
@@ -1586,8 +1586,8 @@ namespace ranges {
             template <class _Rng>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
                 if constexpr (view<remove_cvref_t<_Rng>>) {
-                    if constexpr (constructible_from<remove_cvref_t<_Rng>, _Rng>) {
-                        return {_St::_View, is_nothrow_constructible_v<remove_cvref_t<_Rng>, _Rng>};
+                    if constexpr (convertible_to<_Rng, remove_cvref_t<_Rng>>) {

```

Comment:
Pre-existing: this shouldn't be constrained regardless of the actual constraint. [range.all.general]/2.1 says that `views::all(E)` is expression-equivalent to "`decay-copy(E)` if the decayed type of `E` models `view`", it notably does not require that `decay-copy(E)` is well-formed.

This distinction didn't matter before adding move-only views to the design: `decay-copy(E)` was always well-formed. After, it is possible that the decayed type of `E` is a move-only view but `decay-copy(E)` is ill-formed because `E` is `const` and/or an lvalue. Nevertheless, we don't want to constrain it away: trying to use an lvalue of move-only view type in view composition _should_ be a hard error reported to the user via `static_assert`. We don't want to silently fall back on non-`view` behavior, we want to make it clear to the user that we understand their intent but that it's not possible to make it so.

TLDR: we should `static_assert` this `convertible_to` requirement in the `_St::_View` branch of `operator()` with a nice error message like "views::all cannot accept an lvalue of a move-only view type".

Reply:
> TLDR: we should `static_assert` this `convertible_to` requirement in the `_St::_View` branch of `operator()` with a nice error message like "views::all cannot accept an lvalue of a move-only view type".

It seems that the intent of LWG-3724 is to properly constrain the `operator()`, and hence this suggestion is opposite to such intent...

Reply:
Ah, thanks for the reminder: we need to emulate the result of substitution causing the equivalent expression to be ill-formed.